File: ride-service/src/main/java/com/internship/rideservice/config/ArtemisConfig.java
package com.internship.rideservice.config;

import jakarta.jms.ConnectionFactory;
import jakarta.jms.JMSException;
import org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.jms.core.JmsTemplate;
import org.springframework.jms.support.converter.MappingJackson2MessageConverter;
import org.springframework.jms.support.converter.MessageType;

@Configuration
public class ArtemisConfig {

    @Value(value = "${spring.artemis.broker-url}")
    private String brokerUrl;
    @Value(value = "${spring.artemis.user}")
    private String user;
    @Value(value = "${spring.artemis.password}")
    private String password;

    @Bean
    public ConnectionFactory connectionFactory() throws JMSException {
        ActiveMQConnectionFactory factory = new ActiveMQConnectionFactory();
        factory.setBrokerURL(brokerUrl);
        factory.setUser(user);
        factory.setPassword(password);
        return factory;
    }

    @Bean
    public JmsTemplate jmsTemplate(ConnectionFactory connectionFactory) {
        JmsTemplate jmsTemplate = new JmsTemplate(connectionFactory);
        jmsTemplate.setMessageConverter(jacksonMessageConverter());
        return jmsTemplate;
    }
    @Bean
    public MappingJackson2MessageConverter jacksonMessageConverter() {
        MappingJackson2MessageConverter converter = new MappingJackson2MessageConverter();
        converter.setTargetType(MessageType.TEXT);
        converter.setTypeIdPropertyName("_type");
        return converter;
    }
}

File: finance-service/src/main/kotlin/com/internship/financeservice/config/ArtemisConfig.kt
package com.internship.financeservice.config

import jakarta.jms.ConnectionFactory
import jakarta.jms.JMSException
import org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.jms.core.JmsTemplate
import org.springframework.jms.support.converter.MappingJackson2MessageConverter
import org.springframework.jms.support.converter.MessageType

@Configuration
class ArtemisConfig {
    @Bean
    @Throws(JMSException::class)
    fun connectionFactory(): ConnectionFactory {
        val factory = ActiveMQConnectionFactory()
        factory.setBrokerURL("tcp://localhost:61616")
        factory.setUser("admin")
        factory.setPassword("admin")
        return factory
    }

    @Bean
    fun jmsTemplate(connectionFactory: ConnectionFactory?): JmsTemplate {
        val jmsTemplate = JmsTemplate(connectionFactory!!)
        jmsTemplate.messageConverter = jacksonMessageConverter()
        return jmsTemplate
    }

    @Bean
    fun jacksonMessageConverter(): MappingJackson2MessageConverter {
        val converter = MappingJackson2MessageConverter()
        converter.setTargetType(MessageType.TEXT)
        converter.setTypeIdPropertyName("_type")
        return converter
    }
}